packages/explorer/src/app/(explorer)/[chainName]/worlds/[worldAddress]/explore/utils/getLimitOffset.ts

```diff
@@ -0,0 +1,32 @@
+import { Parser } from "node-sql-parser";
+
+const sqlParser = new Parser();
+const opt = {
+  database: "Postgresql",
+};
+
+export function getLimitOffset(query: string) {
+  const decodedQuery = decodeURIComponent(query);
```

Comment:
same question as above, I am confused why this isn't already decoded

Reply:
It is but if we don't `decodeURIComponent` and `encodeURIComponent`, it loses new line characters inside the query. Found this to work reliably.

Reply:
Actually, a good time to put this into a reusable hook, added `useSQLQueryState.ts`

Reply:
this sounds like `useQueryState` is broken then, using native `URLSearchParams` should correctly encode/decode without any data loss

<img width="571" alt="image" src="https://github.com/user-attachments/assets/5756817d-7250-4066-a5fa-35b141af3699" />